segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -580,6 +620,15 @@ public void mergeSegmentsBatch(WireCommands.MergeSegmentsBatch mergeSegments) {
                });
     }
 
+    private CompletableFuture<Void> appendOnIndexSegment(String segmentName) {
+            return segmentStore.getAttributes(getIndexSegmentName(segmentName), Collections.singleton(EXPECTED_INDEX_SEG_EVENT_SIZE), true, TIMEOUT)
+                    .thenApply(attributes -> attributes.getOrDefault(EXPECTED_INDEX_SEG_EVENT_SIZE, 0L))
+                    .exceptionally(e -> {
+                        log.warn("Exception occured while getting max event size for index segment {}, exception: {}", getIndexSegmentName(segmentName), e);
```

Comment:
Do you think we need to handle **StreamSegmentNotExistsException** here and create segment if doesn't exist. It will help in upgrade scenario.

Reply:
I am not sure what you mean by handle. Do you mean in that case it should log an info instead of a warn? Or do you mean it should be created?

Reply:
I mean if index segment is not available for target stream segment then it will be good if we create index segment here. So that in next **txn.commit** call, will have entry in index segment.